### PR TITLE
Bugfix/fix select and drag

### PR DESCRIPTION
**Pull request by SPConrad**
I was making this a lot more complicated than necessary. 

Added the "show the widget is moving" styling to the class that's already assigned to do that

Check for "is dragging" on mouseUp, if dragging, do not select that widget for editing 

Deselect cell if click off of cells and buttons

Select new cell if unselected cell is clicked

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -99,8 +96,25 @@ const jsonButton = document.getElementById('jsonButton');
...
+var activelyDragging = false; 
+
+function getSelectedElementId() {
+    return selectedElement.parentElement.getAttribute("id").substr(-1); 
+}
+
+const mouseUp = document.addEventListener('mouseup', function(e){
+    const target = e.target
+    if (target.classList.contains("widget-box")) {
+        if (e.type === "mouseup") {
```
Comment (iamsellek) [inline main.js]: Can this be &&ed with the above if statement?

File: main.js [hunk 2]
```diff
@@ -99,8 +96,25 @@ const jsonButton = document.getElementById('jsonButton');
...
+            selectCell(target);  
+        }
+    } else if (target.nodeName != "BUTTON") {
```
Comment (iamsellek) [inline main.js]: * `!==`

Always use `===` and `!==` when making comparisons in JS. `==` and `!=` will try to coerce objects, which can result in unexpected behavior.

File: main.js [hunk 3]
```diff
@@ -437,7 +444,7 @@ function getRowOfLastVisibleCell() {
     const numberOfColumns = columnInput.value;
     for(i = items.length - 1; i >= 0; i--) {
         //New index in order to generate JSON with the new order of elements reflacted
-        let itemID = items[i].getAttribute('id');
+        let itemID = items[i].getAttribute("id").substr(-1);
```
Comment (iamsellek) [inline main.js]: JS generally does single quotes. A linter would have fixed this if this project had one, so I'm assuming it doesn't? Either way, could you fix this one and the one on line 474?

File: main.js [hunk 4]
```diff
@@ -690,36 +695,25 @@ function printObjectProperties(object) {
 
 //Listener that handles the mousedown event on each grid item to be able to set the selectClicked bool
 //selectClicked is used to prevent the grid item from scrolling when the user goes to select a new list value
-function handleListEvent(event) {
-    const target = event.target;
-    const targetType = target.constructor.name;
-
-    //Debugging prints
-    // console.log('event: ' + event.type);
```
Comment (iamsellek) [inline main.js]: Nice removal!

File: main.js (hunk 1 above)
Comment (SPConrad) [reply main.js]: Done

File: main.js (hunk 2 above)
Comment (SPConrad) [reply main.js]: Thanks! I did not know that

File: main.js (hunk 3 above)
Comment (SPConrad) [reply main.js]: 👍

**Review by iamsellek — APPROVED** [review]

**Pull request merged**